packages/core/src/tracing/instrumentation/database/ioredis.js

```diff
@@ -49,18 +76,19 @@ function instrumentSendCommand(original) {
       return original.apply(this, arguments);
     }
 
+    // ".multi()" commands could actually be recorded as multiple spans, but we ONLY want to record ONE
+    // batched span considering that a multi call represents a transaction.
+    // The same is true for pipeline calls, but they have a slightly different semantic.
     if (
       parentSpan.n === exports.spanName &&
-      (parentSpan.data.redis.command === 'multi' || parentSpan.data.redis.command === 'pipeline') &&
-      // the multi call is handled in instrumentMultiCommand but since multi is also send to Redis it will also
-      // trigger instrumentSendCommand, which is why we filter it out.
-      command.name !== 'multi'
+      (parentSpan.data.redis.command === 'multi' || parentSpan.data.redis.command === 'pipeline')
```

Comment:
Here was a bug IMO:

For multi, we only want to capture one span. As the original comment describes it:

> multi commands could actually be recorded as multiple spans, but we only want to record one
   batched span considering that a multi call represents a transaction.
   
But we recorded 3 spans, not one. Because the span length assertion was missing, we did not know this.

I removed

> && command.name !== 'multi'

Which was wrong.

We want to NOT create a span when the parent span is a redis span (the multi/pipeline execution) and the command is MULTI or PIPELINE.
And then we skip recording sub commands if command name is multi or pipeline (line 87)

Reply:
Its really good we added the span count check in our tests 👍